src/univariate/continuous/normal.jl

```diff
@@ -211,16 +211,16 @@ end
 
 # fit_mle based on sufficient statistics
 
-fit_mle(::Type{<:Normal}, ss::NormalStats) = Normal(ss.m, sqrt(ss.s2 / ss.tw))
+fit_mle(::Type{D}, ss::NormalStats) where {D<:Normal} = D(ss.m, sqrt(ss.s2 / ss.tw))
 fit_mle(g::NormalKnownMu, ss::NormalKnownMuStats) = Normal(g.μ, sqrt(ss.s2 / ss.tw))
 fit_mle(g::NormalKnownSigma, ss::NormalKnownSigmaStats) = Normal(ss.sx / ss.tw, g.σ)
 
 # generic fit_mle methods
 
-function fit_mle(::Type{<:Normal}, x::AbstractArray{T}; mu::Float64=NaN, sigma::Float64=NaN) where T<:Real
+function fit_mle(::Type{D}, x::AbstractArray{<:Real}; mu::Float64=NaN, sigma::Float64=NaN) where {D<:Normal}
     if isnan(mu)
         if isnan(sigma)
-            fit_mle(Normal, suffstats(Normal, x))
+            fit_mle(D, suffstats(Normal, x))
         else
             g = NormalKnownSigma(sigma)
```

Comment:
We should find a way to preserve/forward the type of `D` in these cases as well.

Reply:
Which eltype should we pick for `D` though, the one of `g` or the one of the suffstats?

Reply:
I added a conversion in my latest commit, what do you think?